src/ui.rs

```diff
@@ -171,6 +172,18 @@ impl Ui {
         Ok(())
     }
 
+    pub fn write_clap_err_and_exit(&mut self, err: clap::Error) -> ! {
+        // Color parsing must never panic, otherwise we'll never print help.
+        let color = use_color(color_setting(&self.settings));
+        let clap_str = if color {
+            err.render().ansi().to_string()
+        } else {
+            err.render().to_string()
+        };
+        self.write(clap_str.as_str()).ok();
+        std::process::exit(129)
```

Comment:
This should instead return a `CommandError`. The existing `CliError` variant seems appropriate, but that expects an error message (and prints a "Error: " before it). So maybe split that up into two different "CLI" errors?

Reply:
Do we need to return an error if we always exit? (This replaces the panic!-ing clap methods.)

Reply:
It might make sense to report the error (and *then* exit) rather than silently eating it, at least.

Reply:
Exiting here is not a regression, but returning an error and having the exit() call in a single place is a cleanup I was looking forward to. It can be in a separate commit (and even a separate PR) if you prefer.

Reply:
Ah, I see. Sure, that makes sense to do in this PR.